app/models/attachment_reference.rb

```diff
@@ -0,0 +1,61 @@
+class AttachmentReference < ActiveRecord::Base
+  after_initialize :set_expiry_at, if: proc { new_record? && !attachable }
+  before_validation :update_expiry_at
```

Comment:
Do a debug, FactoryGirl should, IIRC.

I'm not sure if implicitly setting an expiry is a good thing to do... why would you prefer fixing the expiry date automatically instead of allowing the user to set it? I want to understand the rationale.